apollo-federation/src/connectors/json_selection/methods/public/lte.rs

```diff
@@ -0,0 +1,372 @@
+use apollo_compiler::collections::IndexMap;
+use serde_json_bytes::Value as JSON;
+use shape::Shape;
+use shape::ShapeCase;
+use shape::location::SourceId;
+
+use crate::connectors::json_selection::ApplyToError;
+use crate::connectors::json_selection::ApplyToInternal;
+use crate::connectors::json_selection::MethodArgs;
+use crate::connectors::json_selection::VarsWithPathsMap;
+use crate::connectors::json_selection::immutable::InputPath;
+use crate::connectors::json_selection::location::Ranged;
+use crate::connectors::json_selection::location::WithRange;
+use crate::impl_arrow_method;
+
+impl_arrow_method!(LteMethod, lte_method, lte_shape);
+/// Returns true if the applied to value is less than or equal to the argument value.
+/// Simple examples:
+///
+/// $(3)->lte(3)       results in true
+/// $(2)->lte(3)       results in true
+/// $(4)->lte(3)       results in false
+/// $("a")->lte("b")   results in true
+/// $("c")->lte("b")   results in false
+fn lte_method(
+    method_name: &WithRange<String>,
+    method_args: Option<&MethodArgs>,
+    data: &JSON,
+    vars: &VarsWithPathsMap,
+    input_path: &InputPath<JSON>,
+) -> (Option<JSON>, Vec<ApplyToError>) {
+    let Some(first_arg) = method_args.and_then(|args| args.args.first()) else {
+        return (
+            None,
+            vec![ApplyToError::new(
+                format!(
+                    "Method ->{} requires exactly one argument",
+                    method_name.as_ref()
+                ),
+                input_path.to_vec(),
+                method_name.range(),
+            )],
+        );
+    };
+
+    let (value_opt, arg_errors) = first_arg.apply_to_path(data, vars, input_path);
+    let mut apply_to_errors = arg_errors;
+    // We have to do this because Value doesn't implement PartialOrd
+    let matches = value_opt.is_some_and(|value| {
+        match (data, &value) {
+            // Number comparisons
+            (JSON::Number(left), JSON::Number(right)) => {
+                left.as_f64().unwrap_or(0.0) <= right.as_f64().unwrap_or(0.0)
```

Comment:
I don't think substituting `0.0` here is a good idea, especially since it returns `true` if both unwrapping fail. Let's return `false` with an error if either conversion fails.